R/getDA.R

```diff
@@ -0,0 +1,812 @@
+    paired = !is.null(pair.by), ...
+) {
+    # Either assay.type. row.var or col.var must be specified
+    if( sum(c(is.null(assay.type), is.null(row.var), is.null(col.var))) != 2L ){
+        stop("Please specify either 'assay.type', 'row.var', or 'col.var'.",
+             call. = FALSE)
+    }
+    # features cannot be specified if row.var or col.var is specified
+    if( is.null(assay.type) && !is.null(features) ){
+        stop("'features' can only be specified when 'assay.type' is ",
+             "specified.", call. = FALSE)
+    }
+    # As features points to rownames, the TreeSE must have rownames and features
+    # must match them
+    if( !is.null(features) && is.null(rownames(tse)) ){
+        stop("'object' must have rownames.", call. = FALSE)
+    }
+    if( !(is.null(features) ||
+          (is.character(features) && all(features %in% rownames(tse)) )) ){
+        stop("'features' must be NULL or single character value specifying ",
+             " rownames.", call. = FALSE)
+    }
+    # If assay was specified, check that it is correct.
+    if( !is.null(assay.type) ){
+        .check_assay_present(assay.type, tse)
+    }
+    # validate method
+    da.method <- .check_method(da.method)
+    # check method-specific requirements
+    if ( da.method == "friedman" && !paired ) {
+        stop("Friedman test requires a 'pair.by' variable.", call. = FALSE)
+    }
+    if ( da.method == "kruskal" && paired ) {
+        stop("Kruskal-Wallis test does not support paired designs.", call. = FALSE)
+    }
+    
+    # Check colData/rowData variables
+    temp <- .check_metadata_variable(tse, row.var, row = TRUE)
+    temp <- .check_metadata_variable(tse, col.var, col = TRUE)
+    temp <- .check_metadata_variable(
+        tse, group,
+        row = length(c(row.var, assay.type))>0,
+        col = length(c(col.var, assay.type))>0
+    )
+    temp <- .check_metadata_variable(
+        tse, comp.by,
+        row = length(c(row.var, assay.type))>0,
+        col = length(c(col.var, assay.type))>0
+    )
+    temp <- .check_metadata_variable(
+        tse, facet.by,
+        row = length(c(row.var, assay.type))>0,
+        col = length(c(col.var, assay.type))>0
+    )
+    temp <- .check_metadata_variable(tse, pair.by, col = TRUE)
+
+    # Check that repeated measures are balanced when 'pair.by' is used
+    if (!is.null(pair.by)) {
+        coldata <- as.data.frame(colData(tse))
+        
+        sample_counts <- coldata %>%
+            group_by(.data[[pair.by]]) %>%
+            summarise(n_samples = n(), .groups = "drop")
+        
+        if (length(unique(sample_counts$n_samples)) > 1) {
+            stop(
+                "When 'pair.by' is specified, all subjects must have the ",
+                "same number of samples.\nRemove 'pair.by' or filter your ",
+                "data to include only subjects with balanced repeats.",
+                call. = FALSE
+            )
+        }
+    }
```

Comment:
Is this required for the function to work? I guess user should already be aware of this, and we should not restrict user to run the method if some samples are missing from certain time points